### PR TITLE
Fixed building on Windows

### PR DESCRIPTION
**Pull request by nefarius**
This PR made all adjustments necessary to build the solution for the Windows platform.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,392 @@
...
+*.cab
+*.msi
+*.msix
+*.msm
+*.msp
+
+# JetBrains Rider
+.idea/
+*.sln.iml
+
```
Comment (jeanthom) [inline .gitignore]: I'd rather prefer to not have a "one size fits all" gitignore. IDE-specific ignored files should ideally go into your global gitignore file ([see this article for more info](https://codebabes.com/blog/set-gitignore-globally-ignore-files-bad-ass))

File: src/strings.h [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+#endif
+
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
+
+#endif /* _WIN32 */
+
+#endif /* __STRINGS_H_868520A6_2F4E_4B49_AF4B_02B5A21CF6BB__ */
```
Comment (jeanthom) [inline src/strings.h]: Isn't that provided by Microsoft C runtime? ~~If not, it shall be moved into `contrib/`.~~

File: usb2sniffer-cli.sln [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+		{666DF0F3-BA65-462C-A3E5-430528ACD8E1}.Release|x86.ActiveCfg = Release|Win32
+		{666DF0F3-BA65-462C-A3E5-430528ACD8E1}.Release|x86.Build.0 = Release|Win32
+	EndGlobalSection
+	GlobalSection(SolutionProperties) = preSolution
+		HideSolutionNode = FALSE
+	EndGlobalSection
+	GlobalSection(ExtensibilityGlobals) = postSolution
+		SolutionGuid = {09F21A6B-F0A2-4AAA-8B1D-CFE8E0A6BA7D}
+	EndGlobalSection
+EndGlobal
```
Comment (jeanthom) [inline usb2sniffer-cli.sln]: Visual Studio files should go in `contrib/`

File: xbar/xbar.c [hunk 4]
```diff
@@ -10,12 +12,27 @@
...
+
+    ft.QuadPart = -(10*usec); // Convert to 100 nanosecond interval, negative value indicates relative time
+
+    timer = CreateWaitableTimer(NULL, TRUE, NULL); 
+    SetWaitableTimer(timer, &ft, 0, NULL, NULL, 0); 
+    WaitForSingleObject(timer, INFINITE); 
+    CloseHandle(timer); 
+}
+#endif
+#include "packing.h"
```
Comment (jeanthom) [inline xbar/xbar.c]: Windows definition of usleep should go after `#include "packing.h"`

File: xbar/xbar.h [hunk 5]
```diff
@@ -1,7 +1,7 @@
 #ifndef __XBAR_H
 #define __XBAR_H
 
-#include <strings.h>
+#include "strings.h"
```
Comment (jeanthom) [inline xbar/xbar.h]: I would favor this instead:

```c++
#ifndef _WIN32
#include <strings.h>
#else
#include "strings-win32.h"
#endif
```

~~Or to add `contrib/` as an include folder for Windows builds~~

File: packing.h [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+#ifdef __GNUC__
+#define PACK( __Declaration__ ) __Declaration__ __attribute__((__packed__))
+#endif
+
+#ifdef _MSC_VER
+#define PACK( __Declaration__ ) __pragma( pack(push, 1) ) __Declaration__ __pragma( pack(pop))
+#endif
```
Comment (jeanthom) [inline packing.h]: I'm not sure how to feel about this. GCC supports MSVC packing pragma so we could avoid doing this, but I'm not sure about clang.

**Review by jeanthom — CHANGES_REQUESTED** [review]
Hi, and thank you for your precious pull-request!

usb2sniffer support in Windows is probably the most demanded feature, and as Linux/Mac user I haven't been able to provide good support for those users, so thanks for taking care of this!

I am myself not a user of Visual Studio; I would appreciate to have VS-specific files into a `contrib/` folder ([some context](https://drewdevault.com/2020/06/06/Add-a-contrib-directory.html)).

Regarding WIN32 quirks, I made a couple of suggestions for better integrating these changes into the code base.

Thanks again!
Jean.

**Comment by delthas** [comment]
It'd be much cleaner to create a CMakeLists file for the project IMO, since Visual Studio supports CMake projects, rather than using the proprietary sln, vcxproj etc files.

**Comment by jeanthom** [comment]
Sure, that could be a better way to unify builds on Windows/Mac/Linux. Is QT well supported?

**Comment by nefarius** [comment]
I'll take care of all the remarks, except for CMakeLists, I have no experience with CMake.

File: .gitignore (hunk 1 above)
Comment (nefarius) [reply .gitignore]: Totally makes sense, now you will laugh though; VS doesn't seem to use/respect my global .gitignore settings so until that is fixed I trimmed down the gitignore so I don't accidentally commit garbage 😅

File: src/strings.h (hunk 2 above)
Comment (nefarius) [reply src/strings.h]: Haha, oops, it is, I'll fix and remove that.

**Comment by jeanthom** [comment]
Thanks again for the PR, @nefarius. Like you I'm not familiar enough with CMake, I prefer to integrate your VS solution into my repo instead.

Regarding the global gitignore issue, that sucks pretty bad :( A workaround would be to use ".git/info/exclude", which isn't part of the repository.

**Comment by nefarius** [comment]
No worries, I'll sure find a way, thanks for the informative links and accepting the PR! This is currently the only tool actually usable for HS capture since the Qt application instantly crashes on capture (at least for me) and all I want is a PCAP file I can analyze in peace afterwards anyway. I tried to take a spin on fixing the Qt app but honestly, Qt being so insanely developer hostile lately, requiring an account to even download their build tools... No thank you, rather use the CLI and Wireshark instead.

**Pull request merged**